pkg/credentialprovider/plugin/config.go

```diff
@@ -43,6 +45,15 @@ func readCredentialProviderConfigFile(configPath string) (*kubeletconfig.Credent
 		return nil, fmt.Errorf("error decoding config %s: %w", configPath, err)
 	}
 
+	// Append current system environment variables, to the ones configured in the
```

Comment:
Config file is read during registration of credential providers i.e during Kubelet start, any changes later to OS env variable will not be picked by exec plugin during execution, any reasons to not to do it at original place? https://github.com/kubernetes/kubernetes/blob/master/pkg/credentialprovider/plugin/plugin.go#L348

Reply:
@adisky Sure makes sense, though my rationale was providing a `CredentialProviderConfig` is the job of `readCredentialProviderConfigFile` and all other functions are consumers of it.

If I am appending env vars in the `ExecPlugin` function then it's implied that `ExecPlugin` knows that env vars could be stale and this breaks the abstraction between `ExecPlugin` and `readCredentialProviderConfigFile`

At the moment, `ExecPlugin` does is just take the env vars present in the `CredentialProviderConfig` and execute it and I didn't want to break this contract.

I'll wait for the feedback from the other reviewers and if general consensus is to move it to `ExecPlugin` to inject latest env vars without having to perform a restart of the kubelet then I'll do it.

Reply:
@n4j I don't have any hard views here, was interested in rationale behind doing it here.

Reply:
I would not recommend mutating the read config here, but would construct this at invocation time.

I like the approach taken by client-go exec plugin that stubs an `environ func() []string` for testing and defaults it to `os.Environ`

I would also expect the envvars in the config file to be appended, which makes them take precedence over the os.Environ